src/wallet/wallet.cpp

```diff
@@ -5683,7 +5683,9 @@ bool CWallet::GenerateNewHDChain(const SecureString& secureMnemonic, const Secur
 
         // We got a gibberish key...
         if (vMasterKey.empty()) {
-            throw std::runtime_error(strprintf("%s: supplied incorrect passphrase", __func__));
+            // Mimicking the error message of RPC_WALLET_PASSPHRASE_INCORRECT as it's possible
```

Comment:
you can actually write try/catch inside rpc code and if the exception - is 'passphrase is incorrect' - re-throw it as an RPCError. 

I think that would be the best from the rpc's user point of view.

Reply:
Seems a bit convoluted, it'll help us retain the right error code but catching-processing-rethrowing been done anywhere else?

Reply:
This makes the user-facing error look like "Generating new chain failed due to: GenerateNewHDChain: supplied incorrect passphrase" (assume we restore the pre-`RPC_WALLET_PASSPHRASE_INCORRECT` error message).

If trying to ensure the incorrect passphrase error has the correct error code, it'd look a bit like this

```c++
    bool wallet_upgraded{false};
    try {
        wallet_upgraded = pwallet->UpgradeToHD(secureMnemonic, secureMnemonicPassphrase, secureWalletPassphrase, error);
    } catch (const std::runtime_error& e) {
        if (e.what() == std::string("supplied incorrect passphrase") {
            throw JSONRPCError(RPC_WALLET_PASSPHRASE_INCORRECT, "Error: The wallet passphrase entered was incorrect");
        }
        // No more exceptions to translate, rethrow them
        throw std::runtime_error(e.what());
    } // Let other exceptions remain unhandled by us
```

But this feels like excessive effort when the alternative (just mirroring the error message we expect from `RPC_WALLET_PASSPHRASE_INCORRECT`) seems fine.